sapl/sessao/views.py

```diff
@@ -4384,6 +4384,90 @@ class DeleteView(MasterDetailCrud.DeleteView):
+                                                retiradapauta=None, tipo_votacao=4, registroleitura__materia=None)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['pk'] = self.kwargs['pk']
+        context['root_pk'] = self.kwargs['pk']
+        if not verifica_sessao_iniciada(self.request, self.kwargs['pk']):
+            context['sessao_iniciada'] = False
+            return context
+        context['sessao_iniciada'] = True
+        context['turno_choices'] = Tramitacao.TURNO_CHOICES
+        context['title'] = SessaoPlenaria.objects.get(id=self.kwargs['pk'])
+        if self.expediente:
+            context['expediente'] = True
+        else:
+            context['expediente'] = False
+        return context
+
+    def post(self, request, *args, **kwargs):
+        if 'marcadas_4' in request.POST:
+            models = None
+            if request.POST['origem'] == 'ordem':
+                models = OrdemDia.objects.filter(id__in=request.POST.getlist('marcadas_4'))
+            elif request.POST['origem'] == 'expediente':
+                models = ExpedienteMateria.objects.filter(id__in=request.POST.getlist('marcadas_4'))
+
+            if models is None:
+                messages.add_message(self.request, messages.ERROR,
+                                     _('Impossível localizar as matérias selecionadas'))
+                return self.get(request, self.kwargs)
+            else:
+                for m in models:
+                    try:
+                        obj = None
+                        if isinstance(m, ExpedienteMateria):
+                            obj = RegistroLeitura.objects.filter(materia=m.materia, expediente=m).delete()
+                            obj = RegistroLeitura()
+                            obj.expediente = m
+                        elif isinstance(m, OrdemDia):
+                            obj = RegistroLeitura.objects.filter(materia=m.materia, ordem=m).delete()
+                            obj = RegistroLeitura()
+                            obj.ordem = m
+                        else:
+                            messages.add_message(self.request, messages.ERROR,
+                                                 _('Tipo de Pauta nao encontrado'))
+                            return self.get(request, self.kwargs)
+
+                        obj.materia = m.materia
+                        obj.observacao = request.POST['observacao']
+                        obj.user = self.request.user
+                        obj.ip = get_client_ip(self.request)
+                        obj.save()
+
+                    except Exception as e:
+                        messages.add_message(self.request, messages.ERROR,
+                                             _('Erro ao salvar registro de Leitura, por favor, refaça a operação'))
+                        return self.get(request, self.kwargs)
+                    else:
```

Comment:
Particularmente, acho esses `else` depois de `try-except` ou depois de `for loops` propícios a erros. Por que não colocar as linhas 4453 a 4455 após a linha 4446? O que se ganha definindo isso como um `else` a parte? Eu acho que esse `else` após try-except é válido quando precisamos limpar resources tipo conexão de BD, invalidação de cache, etc.